server/routes/contacts/create/enter-dob/createContactEnterDobSchemas.test.ts

```diff
@@ -0,0 +1,225 @@
+      // Given
+      const form = { isDobKnown: 'true', day: '1', month, year: '1999' }
+
+      // When
+      const result = await doValidate(form)
+
+      // Then
+      expect(result.success).toStrictEqual(isValid)
+      if (!isValid) {
+        const deduplicatedFieldErrors = deduplicateFieldErrors(result)
+        expect(deduplicatedFieldErrors).toStrictEqual({
+          month: ['Enter a valid month (1-12)'],
+        })
+      }
+    })
+    it.each([
+      ['0', false],
+      ['-1', false],
+      ['1899', false],
+      ['1900', true],
+    ])('if dob is known then year must be at least 1900 (%s, %s)', async (year: string, isValid: boolean) => {
+      // Given
+      const form = { isDobKnown: 'true', day: '1', month: '1', year }
+
+      // When
+      const result = await doValidate(form)
+
+      // Then
+      expect(result.success).toStrictEqual(isValid)
+      if (!isValid) {
+        const deduplicatedFieldErrors = deduplicateFieldErrors(result)
+        expect(deduplicatedFieldErrors).toStrictEqual({
+          year: ['Enter a valid year. Must be at least 1900'],
+        })
+      }
+    })
+    it('dob must not be in the future', async () => {
+      // Given
+      const form = { isDobKnown: 'true', day: '1', month: '1', year: '2045' }
+
+      // When
+      const result = await doValidate(form)
+
+      // Then
+      expect(result.success).toStrictEqual(false)
+      const deduplicatedFieldErrors = deduplicateFieldErrors(result)
+      expect(deduplicatedFieldErrors).toStrictEqual({
+        isDobKnown: ['The date of birth must not be in the future'],
+      })
+    })
+    it.each([
+      ['01', '06', '1982', new Date('1982-06-01T00:00:00.000Z')],
+      ['1', '6', '1982', new Date('1982-06-01T00:00:00.000Z')],
+    ])('dob should parse to a date correctly', async (day: string, month: string, year: string, expected: Date) => {
+      // Given
+      const form = { isDobKnown: 'true', day, month, year }
+
+      // When
+      const result = await doValidate(form)
+
+      // Then
+      expect(result.success).toStrictEqual(true)
+      expect(result.data).toStrictEqual({
+        isDobKnown: true,
+        dateOfBirth: expected,
+      })
+    })
+    it('should not map dob if not known', async () => {
+      // Given - we can select yes and enter day, month and year and then change to no so they may be populated in the form
+      const form = { isDobKnown: 'false', day: '1', month: '1', year: '1900' }
+
+      // When
+      const result = await doValidate(form)
+
+      // Then
+      expect(result.success).toStrictEqual(true)
+      expect(result.data).toStrictEqual({
+        isDobKnown: false,
+      })
+    })
+    it('should handle empty/invalid dob fields if dob is not known', async () => {
+      // Given - we can select yes and enter day, month and year and then change to no so they may be populated in the form
```

Comment:
is this comment necessary here, if it is may be update the test method description to reflect this ?